File: src/main/java/com/windchopper/fs/SftpFile.java
package com.windchopper.fs;

import com.jcraft.jsch.SftpATTRS;

public class SftpFile implements SftpConstants {

    final String path;
    final String name;
    final SftpATTRS attributes;

    SftpFile(String path, String name, SftpATTRS attributes) {
        this.path = path;
        this.name = name;
        this.attributes = attributes;
    }

    String toAbsolutePath() {
        return (path.equals(SEPARATOR) ? "" : path) + SEPARATOR + name;
    }

    SftpPath toPath(SftpFileSystem fileSystem, SftpSessionIdentity sessionIdentity) {
        return new SftpPath(fileSystem, sessionIdentity, path, name);
    }

    SftpFileAttributes toFileAttributes() {
        return new SftpFileAttributes(attributes);
    }

}

File: src/main/kotlin/com/github/windchopper/fs/sftp/SftpFile.kt
package com.github.windchopper.fs.sftp

import com.jcraft.jsch.SftpATTRS

class SftpFile(val path: String, val name: String, val attributes: SftpATTRS) {

    fun toAbsolutePath(): String {
        return (if (path == SftpFileSystem.PATH_SEPARATOR) "" else path) + SftpFileSystem.PATH_SEPARATOR + name
    }

    fun toPath(fileSystem: SftpFileSystem, sessionIdentity: SftpConfiguration.SessionIdentity): SftpPath {
        return SftpPath(fileSystem, sessionIdentity, path, name)
    }

    fun toFileAttributes(): SftpFileAttributes {
        return SftpFileAttributes(attributes)
    }

}